src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        putValue(hash(key), key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Node<K, V> newNode = table[index];
+        while (newNode != null) {
+            if (Objects.equals(key, newNode.key)) {
+                return newNode.value;
+            }
+            newNode = newNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(Object key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private void putValue(int hash, K key, V value) {
+        Node<K, V> newNode;
+        if (table[hash] == null) {
+            table[hash] = new Node<>(key, value, null);
+            return;
+        }
+        newNode = table[hash];
+        while (newNode.next != null) {
+            if (Objects.equals(key, newNode.key)) {
+                break;
+            }
+            newNode = newNode.next;
+        }
+        if (Objects.equals(key, newNode.key)) {
+            newNode.value = value;
+            size--;
+            return;
+        }
+        newNode.next = new Node<>(key, value, null);
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        if (size >= threshold) {
+            threshold = (int) (newCapacity * DEFAULT_LOAD_FACTOR);
+            Node<K, V>[] oldTable = table;
+            table = new Node[newCapacity];
+            transfer(oldTable);
+        }
+    }
+
+    private void transfer(Node<K, V>[] oldTable) {
+        for (Node<K, V> node : oldTable) {
+            Node<K, V> newNode = node;
+            while (newNode != null) {
+                putValue(hash(newNode.key), newNode.key, newNode.value);
+                newNode = newNode.next;
+            }
```

Comment:
there is no need to use put() method that checks if key is in table when we already know that all keys are unique
lets rewrite this logic to just fill newTable with old nodes

Reply:
<img width="1280" alt="Screenshot 2022-12-21 at 21 04 00" src="https://user-images.githubusercontent.com/103888898/208983603-8a115677-eee5-461e-97d2-f828d64b98a0.png">
I received an error when null while the check is deleted. So I leave it as it was.